### PR TITLE
konnectivity-server should make the socket group readable and writeab…

### PR DESCRIPTION
**Pull request by vinayakankugoyal**
…le when it listens on the socket. Most systems have a default umask of 022, meaning when konnectivity-server creates the socket only the user that it is running as will have read-write permissions to the socket. This means that any other process that needs to read-write to the socket needs to be run as the same user as konnectivity-server.

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -415,10 +415,12 @@ func (p *Proxy) runUDSMasterServer(ctx context.Context, o *ProxyRunOptions, s *s
 		grpcServer := grpc.NewServer()
 		client.RegisterProxyServiceServer(grpcServer, s)
 		var lc net.ListenConfig
+		oldUmask := syscall.Umask(0007)
 		lis, err := lc.Listen(ctx, "unix", o.udsName)
 		if err != nil {
 			return nil, fmt.Errorf("failed to listen(unix) name %s: %v", o.udsName, err)
 		}
+		syscall.Umask(oldUmask)
```
Comment (caesarxuchao) [inline cmd/server/main.go:423]: This isn't going to be reached if the Listen fails.

**Review by caesarxuchao — COMMENTED** [review]
Thanks, @vinayakankugoyal.

For future reference, this is the recommended way of setting UDS file descriptor's file permission, according to https://github.com/golang/go/issues/11822

File: cmd/server/main.go (hunk 1 above)
Comment (vinayakankugoyal) [reply cmd/server/main.go]: :man_facepalming: Fixed. Thanks for pointing this out.

**Comment by caesarxuchao** [comment]
lgtm. Please squash the commits.

**Comment by vinayakankugoyal** [comment]
> lgtm. Please squash the commits.

Done. Thanks!

**Comment by caesarxuchao** [comment]
LGTM.

I'll defer it to Walter to apply the label.

File: cmd/server/main.go [hunk 2]
```diff
@@ -431,7 +433,9 @@ func (p *Proxy) runUDSMasterServer(ctx context.Context, o *ProxyRunOptions, s *s
 		stop = func() { server.Shutdown(ctx) }
 		go func() {
 			var lc net.ListenConfig
+			oldUmask := syscall.Umask(0007)
```
[the remark this answers is not in the document]
Comment (vinayakankugoyal) [reply cmd/server/main.go]: Strongly agree we should do what you suggested. Updated the code PTAL.

File: cmd/server/main.go [hunk 3]
```diff
@@ -397,6 +400,18 @@ func SetupSignalHandler() (stopCh <-chan struct{}) {
 	return stop
 }
 
+func getUDSListener(ctx context.Context, udsName string) (net.Listener, error) {
+	udsListenerLock.Lock()
+	defer udsListenerLock.Unlock()
+	defer syscall.Umask(syscall.Umask(0007))
```
Comment (cheftako) [inline cmd/server/main.go:406]: Seems a bit sneaky to rely on parameter evaluation of a deferred function to get the first Umask call to happen before the rest of the function. Going to lgtm but we should consider either comments or breaking it out to make it a little more obvious to anyone reading the code.

Comment (vinayakankugoyal) [reply cmd/server/main.go]: Broke it out to make the code more obvious. Thanks!

**Pull request merged**